app/models/amazon_agent_manager.rb

```diff
@@ -0,0 +1,393 @@
+require 'aws-sdk'
+require 'amazon_ssa_support'
+require 'yaml'
+require 'open3'
+
+class AmazonAgentManager
+  include Comparable
+  include Vmdb::Logging
+  attr_accessor :ems, :deploying
+
+  MIQ_SSA   = "MIQ_SSA".freeze
+  RUBY_VERS = "2.3.3".freeze
+  AGENT_AMI = "centos-7.2-hvm*".freeze
+
+  def initialize(ems)
+    @ems = ems
+    @guid = MiqServer.my_server.guid
+
+    # List of active agent ids
+    @alive_agent_ids = []
+
+    # List of all agent ids, include those in power off state.
+    @agent_ids = []
+    @deploying = false
+  end
+
+  def ec2
+    @ec2 ||= ems.connect(:service => 'EC2')
+  end
+
+  def sqs
+    @sqs ||= ems.connect(:service => 'SQS')
+  end
+
+  def s3
+    @s3 ||= ems.connect(:service => 'S3')
+  end
+
+  def iam
+    @iam ||= ems.connect(:service => 'IAM')
```

Comment:
What are the iams for?

Reply:
We need it to create a role/profile. When our agent takes this role/profile, it doesn't need further authentication to run AWS API calls.